### PR TITLE
CodeQL Workflow

### PR DESCRIPTION
**Pull request by AbhijithGanesh**
Addition of yml files for CodeQL workflow checking.

### CODE REVIEW

**Comment by gorkemarslan** [comment]
@AbhijithGanesh, you change a `models.py` file so could you add the migrations that will come with the command of `python manage.py makemigrations`?

**Comment by AbhijithGanesh** [comment]
Yes I can but that shall repeat for other parts too and I shall branch it in a better way. Here codeql and apideprecation got mixed up. I shall fix the same asap.

**Comment by gorkemarslan** [comment]
I think we should not approve code changes in `models.py` files if related migrations are not added. 
But I would say you might need to remove api-deprecation commits from this PR and we approve workflow commits only.
For api-deprecation commits, you can attach migrations to your code changes in a new PR. (Or in the one you just closed.)
We can wait for @brylie, they better know it.

File: project/threads/models.py [hunk 1]
```diff
@@ -1,3 +1,543 @@
...
+
+
+class Fact(models.Model):
+    body = models.CharField(max_length=511)
+
+    created = models.DateTimeField(auto_now_add=True, blank=True, null=True)
+    last_modified = models.DateTimeField(auto_now=True, blank=True, null=True)
+
+
+class Category(models.Model):
```
Comment (gorkemarslan) [inline project/threads/models.py:24]: In issue #1032,  @brylie mentioned that 

> category - although this might be better in a categories app since it is shared between threads and accounts apps

You may not add `Category` model to `threads/models.py`.

**Comment by brylie** [comment]
This pull request lacks context. What is CodeQL and how do we use it? At the very least we should have an open issue to discuss the benefits of this tool and possible alternatives. We may also need to update our project documentation with a note about the tool and its usage.

File: project/threads/models.py [hunk 2]
```diff
@@ -1,3 +1,543 @@
+import os
```
Comment (brylie) [inline project/threads/models.py:1]: As @gorkemarslan mentions, these changes should be in a separate pull request, since they are not related to the CodeQL workflow.

**Review by brylie — CHANGES_REQUESTED** [review]

**Comment by brylie** [comment]
Closing this pull request pending further discussion via a [Feature Request](https://github.com/CiviWiki/OpenCiviWiki/issues/new?assignees=&labels=community%2Ctriage%2Chelp+wanted&template=feature.yml&title=%3CAdd+a+brief+title+for+your+idea%3E+) or [Discussion topic](https://github.com/CiviWiki/OpenCiviWiki/discussions/new).

**Comment by brylie** [comment]
For what it's worth, the CodeQL seems like a good addition. I would just like to explore the enhancement a bit further via discussion :smiley:
